src/OMSimulatorLib/TLM/SystemTLM.cpp

```diff
@@ -419,6 +426,46 @@ oms_status_enu_t oms::SystemTLM::setPositionAndOrientation(const oms::ComRef &cr
   return oms_status_ok;
 }
 
+oms_status_enu_t oms::SystemTLM::fetchInterfaces(const oms::ComRef &cref,
+                                                 std::vector<std::string> &names,
+                                                 std::vector<int> &dimensions,
+                                                 std::vector<std::string> &domains)
+{
+    if(getComponents().find(cref) == getComponents().end()) {
+      return logError("Sub-model \""+std::string(cref)+"\" not found.");
+    }
+    singleModel = cref.c_str();
+    initialize();
+    singleModel = "";
```

Comment:
Is `singleModel` any useful outside of the initialize method?

Reply:
No, but `initialize` is a virtual function, so I cannot add an argument to it without bloating all the other classes with the same argument.